opus/application/static_media/js/browse.js

```diff
@@ -2145,21 +2150,30 @@ var o_browse = {
         let width = o_browse.calculateGalleryWidth(view);
         let height = o_browse.calculateGalleryHeight(view);
 
-        let trCount = viewNamespace.galleryBoundingRect.tr;
+        let trCountFloor = viewNamespace.galleryBoundingRect.trFloor;
         if (!o_browse.isGalleryView(view) && $(`${tab} .op-data-table tbody tr[data-obs]`).length > 0) {
-            trCount = o_utils.floor((height-$("th").outerHeight())/$(`${tab} .op-data-table tbody tr[data-obs]`).outerHeight());
+            trCountFloor = o_utils.floor((height-$("th").outerHeight()) /
+                                         $(`${tab} .op-data-table tbody tr[data-obs]`).outerHeight());
         }
 
         let xCount = o_utils.floor(width/o_browse.imageSize);
-        let yCount = Math.round(height/o_browse.imageSize);
+        // This is the total number of rows that are at least partially visible
+        let yCountCeil = Math.ceil(height/o_browse.imageSize);
+        // This is the total number of rows that are completely visible
+        let yCountFloor = Math.floor(height/o_browse.imageSize);
+        // This is the total number of rows that are visible as much as we think is visually useful
+        let yCountPartial = Math.floor((height + o_browse.imageSize*(1-o_browse.galleryImageViewableFraction)) /
+                                       o_browse.imageSize);
         // let yCount = Math.ceil(height/o_browse.imageSize);
```

Comment:
Do we want to use floor and ceil functions in utils.js? (because height could be a floating point value depending on window's height)

Reply:
Changed these and also a few other cases I found in browse.js